cudamapper/src/index_gpu_two_indices.cuh

```diff
@@ -52,14 +52,14 @@ public:
     /// \param past_the_last_read_id read_id+1 of the last read to be included in this index
     /// \param kmer_size k - the kmer length
     /// \param window_size w - the length of the sliding window used to find sketch elements
-    IndexGPUTwoIndices(io::FastaParser* parser,
+    IndexGPUTwoIndices(const io::FastaParser& parser,
                        const read_id_t first_read_id,
                        const read_id_t past_the_last_read_id,
                        const std::uint64_t kmer_size,
                        const std::uint64_t window_size);
 
     /// \brief Constructor
-    IndexGPUTwoIndices();
+    IndexGPUTwoIndices() = default;
```

Comment:
This default constructor does not make much sense, since the class has `const` data members. I suggest to remove it completely until we needed. Opinions @tijyojwad @vellamike @mimaric ?

Reply:
This would mean
https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/4aac945c3de8d6235ad080fb1e34dba531ee343f/cudamapper/src/index_two_indices.cu#L34
should either be removed as well or return a unique_ptr(nullptr).

Reply:
I agree. Removing it completely.

Reply:
done

Reply:
perfect